src/AutoRest.CSharp/Common/Generation/Writers/ModelWriter.cs

```diff
@@ -287,6 +287,12 @@ private string GetAbstract(ObjectType schema)
 
         protected virtual void AddClassAttributes(CodeWriter writer, ObjectType schema)
         {
+            writer.WriteDeprecatedAttributeIfExists(schema.Deprecated);
+        }
+
+        private void AddClassAttributes(CodeWriter writer, EnumType enumType)
```

Comment:
Can we combine those two methods as one?
AddClassAttributes(CodeWriter writer, TypeProvider schema)

Reply:
`protected virtual void AddClassAttributes(CodeWriter writer, ObjectType schema)` is overridden by https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Generation/ReferenceTypeWriter.cs#L20

That's why I add a separate `AddClassAttribute(CodeWriter writer, EnumType enumType` which is for enum types only, to prevent the behavior for enum types is overridden accidentally.